### PR TITLE
fix(log): show bind parameters in logs instead of placeholders

### PR DESCRIPTION
**Pull request by supperchong**
Fixes https://github.com/sequelize/sequelize/issues/11151
Fixes https://github.com/sequelize/sequelize/issues/10067#

### Pull Request check-list

_Please make sure to review and check all of these items:_

- [x] Does `npm run test` or `npm run test-DIALECT` pass with this change (including linting)?
- [x] Does the description below contain a link to an existing issue (Closes #[issue]) or a description of the issue you are solving?
- [x] Have you added new tests to prevent regressions?
- [ ] Is a documentation update included (if this change modifies existing APIs, or introduces new ones)?
- [ ] Did you update the typescript typings accordingly (if applicable)?
- [x] Did you follow the commit message conventions explained in [CONTRIBUTING.md](https://github.com/sequelize/sequelize/blob/master/CONTRIBUTING.md)?

### CODE REVIEW

**Comment by papb** [comment]
Thanks! Some tests are failing though, can you take a look?

**Comment by supperchong** [comment]
@papb  Yeah!Thanks for your review, I will  fix it.

**Comment by papb** [comment]
Tests are still failing, but you probably know that. Just saying in case you missed it. (Take your time! No rush!)

**Comment by supperchong** [comment]
@papb Thanks for your caring!:smile:

**Comment by papb** [comment]
We now have two equivalent PRs, what to do? Perhaps you two can join forces instead?

#11159 #11238

File: lib/dialects/abstract/query.js [hunk 1]
```diff
@@ -86,7 +86,10 @@ class AbstractQuery {
 
     sql = sql.replace(/\$(\$|\w+)/g, (match, key) => {
       if ('$' === key) {
-        return options.skipUnescape ? match : key;
+        if (!['postgres', 'sqlite'].includes(dialect)) {
+          return options.skipUnescape ? match : key;
+        } 
+        return match;
```
Comment (papb) [inline lib/dialects/abstract/query.js]: Hello, just out of curiosity, why was this change necessary? (I'm a maintainer but I'm sure you understand this part much better than I do)

File: test/integration/model.test.js [hunk 2]
```diff
@@ -934,7 +934,7 @@ describe(Support.getTestDialectTeaser('Model'), () => {
               if (dialect === 'mssql') {
                 expect(sql).to.not.contain('createdAt');
               } else {
-                expect(sql).to.match(/UPDATE\s+[`"]+User1s[`"]+\s+SET\s+[`"]+secretValue[`"]=(\$1|\?),[`"]+updatedAt[`"]+=(\$2|\?)\s+WHERE [`"]+id[`"]+\s=\s(\$3|\?)/);
+                expect(sql).to.match(/Executing \(default\):\sUPDATE\s+[`"]+User1s[`"]+\s+SET\s+[`"]+secretValue[`"]='?(\d+)'?,[`"]+updatedAt[`"]+=('?\d{4}-\d{2}-\d{2}\s\d{2}:\d{2}:\d{2}(\.\d{3})?(\s\+\d{2}:\d{2})?'?)\s+WHERE [`"]+id[`"]+\s=\s(\d+)/);
```
Comment (papb) [inline test/integration/model.test.js]: I believe you could remove this part `Executing \(default\):\s` in the beginning and the regex would still match, right? If yes, please do it.

File: test/integration/model/create.test.js [hunk 3]
```diff
@@ -931,7 +931,26 @@ describe(Support.getTestDialectTeaser('Model'), () => {
...
+          myvals: { type: Sequelize.ARRAY(Sequelize.INTEGER) },
+          mystr: { type: Sequelize.ARRAY(Sequelize.STRING) }
+        });
+
+        let test = false;
+        return User.sync({ force: true }).then(() => {
+          return User.create({ myvals: [1, 2, 3], mystr: ['One', 'Two'] }, {
+            logging(sql) {
+              test = true;
+              expect(sql).to.contain('INSERT INTO "UserWithArrays" ("id","myvals","mystr","createdAt","updatedAt") VALUES (DEFAULT,ARRAY[1,2,3],ARRAY[\'One\',\'Two\']');
```
Comment (papb) [inline test/integration/model/create.test.js]: Here you are checking an incomplete SQL (abruptly ended at `ARRAY[\'One\',\'Two\']`), I think this could cause confusion. Please assert on the complete SQL. I suggest updating the `User` model you create in this test to not have timestamps (`{ timestamps: false }`) so that you don't have to go over the hassle of asserting the `createdAt`/`updatedAt` fields.

File: test/integration/model/create.test.js [hunk 4]
```diff
@@ -942,15 +961,15 @@ describe(Support.getTestDialectTeaser('Model'), () => {
           return User.create({ myvals: [], mystr: [] }, {
             logging(sql) {
               test = true;
-              expect(sql).to.contain('INSERT INTO "UserWithArrays" ("id","myvals","mystr","createdAt","updatedAt") VALUES (DEFAULT,$1,$2,$3,$4)');
+              expect(sql).to.contain('INSERT INTO "UserWithArrays" ("id","myvals","mystr","createdAt","updatedAt") VALUES (DEFAULT,\'{}\',\'{}\'');
```
Comment (papb) [inline test/integration/model/create.test.js]: Same thing as above, please assert on the complete SQL, avoid timestamps with `{ timestamps: false }` in the model.

File: test/integration/model/create.test.js [hunk 5]
```diff
@@ -964,7 +983,7 @@ describe(Support.getTestDialectTeaser('Model'), () => {
             return user.save({
               logging(sql) {
                 test = true;
-                expect(sql).to.contain('UPDATE "UserWithArrays" SET "myvals"=$1,"mystr"=$2,"updatedAt"=$3 WHERE "id" = $4');
+                expect(sql).to.contain('UPDATE "UserWithArrays" SET "myvals"=\'{}\',"mystr"=\'{}\'');
```
Comment (papb) [inline test/integration/model/create.test.js]: Same thing as above, please assert on the complete SQL, avoid timestamps with `{ timestamps: false }` in the model.

**Comment by sushantdhiman** [comment]
Do we require all this, I think we can just append parameters after queries. I have seen this format used with bind queries before

```sql
Executing(default): SELECT * FROM "table" WHERE "name" = ?; test;
```

File: lib/dialects/abstract/query.js (hunk 1 above)
Comment (supperchong) [reply lib/dialects/abstract/query.js]: Because $ is used to escape a literal $ sign. If we unescape `$$1`  with `$1` , then later we replace logSql with bind Parameter, the  `$1` will be `${Parameter}`, but actual expect `$1`.
 For example we have a query   
`sequelize.query('SELECT *, "text with literal $$1 and literal $$status" as t FROM projects WHERE status = $1',
  { bind: ['active'], type: sequelize.QueryTypes.SELECT }
)`  
the log sql will be  
`SELECT *, "text with literal 'active' and literal $status" as t FROM projects WHERE status = 'active'`
but expect   
`SELECT *, "text with literal $1 and literal $status" as t FROM projects WHERE status = 'active'`.

File: test/integration/model.test.js (hunk 2 above)
Comment (supperchong) [reply test/integration/model.test.js]: Yeah, great.

File: test/integration/model/create.test.js (hunk 3 above)
Comment (supperchong) [reply test/integration/model/create.test.js]: Good idea!

**Comment by supperchong** [comment]
@sushantdhiman  Yeah.this would be more easier and uniform. Is there a plan to do this?   
Moreover there is  a little trouble that people should know how module(like [pg](https://github.com/brianc/node-postgres)) hanlde parameters occasionally.

**Comment by papb** [comment]
This PR looks ready to merge for me! :grimacing: Great work, @supperchong!

@sushantdhiman That makes sense, I dunno why we didn't think of that before. Regardless, I think this PR is great as is awyway. I think we should merge it, and perhaps on another occasion we can change back to your suggestion, in another PR. What do you think?

**Comment by sushantdhiman** [comment]
@papb This PR is a good effort but I think we should not merge this because 1) There is new dependency on `sql-string` now 2) There is lots of extra code just to display query logs.

@supperchong You should pass [`_logQuery`](https://github.com/sequelize/sequelize/blob/f3b5f99b342cb5d47865daa49a2aa6316064507c/lib/dialects/abstract/query.js#L330) a third argument `parameters`. Then in that function just append parameters to query if they are not empty. You may continue in this PR or start a new one.

**Comment by papb** [comment]
> 1) There is new dependency on sql-string

This is not true, is it? This is true for the other PR (#11159), not this one.

**Comment by papb** [comment]
Regardless, I think I am convinced. Your idea is simpler and more uniform. Let's just find a good way to show the bound parameters, how would named parameters be handled?

**Comment by sushantdhiman** [comment]
Ah, sorry @papb but still we can't merge this extra code which we really don't need.

**Comment by papb** [comment]
@supperchong Are you willing to rework the PR to follow the new suggestion from sushantdhiman instead of your original idea?

**Comment by supperchong** [comment]
@papb  Yeah, I give another [PR](https://github.com/sequelize/sequelize/pull/11267).  Thanks very much!

**Comment by papb** [comment]
Closing in favor of #11267.
